client/fingerprint_manager.go

```diff
@@ -100,15 +100,20 @@ func (fm *FingerprintManager) setupDrivers(drivers []string) error {
 func (fm *FingerprintManager) fingerprint(name string, f fingerprint.Fingerprint) (bool, error) {
 	request := &cstructs.FingerprintRequest{Config: fm.getConfig(), Node: fm.node}
 	var response cstructs.FingerprintResponse
-	if err := f.Fingerprint(request, &response); err != nil {
+
+	fm.nodeLock.Lock()
+	err := f.Fingerprint(request, &response)
+	fm.nodeLock.Unlock()
+
+	if err != nil {
 		return false, err
 	}
 
-	fm.nodeLock.Lock()
```

Comment:
why is fm.updateNode no longer in the nodelock's scope?

Reply:
IIUC this is fine because the updated node is assigned to the fingerprint manager's node using the lock